dyno-contrib/src/main/java/com/netflix/dyno/contrib/EurekaHostsSupplier.java

```diff
@@ -127,7 +134,7 @@ public String getApplicationName() {
         return applicationName;
     }
 
-    public DiscoveryClient getDiscoveryClient() {
+    public EurekaClient getDiscoveryClient() {
```

Comment:
Even if they assign to `EurekaClient` it could cause problems if someone just gets the new binary without recompiling. The return type is part of the signature used for looking up the method.